File: Ruby/problem1.rb

def sum_of_multiples_below(n, m1 = 3, m2 = 5)
  sum = 0
  for i in 1...n do
    sum += i if i % m1 === 0 || i % m2 === 0
  end
  sum
end

max_plus_one = ARGV.first ? ARGV.first.to_i : 10
start_time = Time.now
puts sum_of_multiples_below(max_plus_one)
puts "#{Time.now - start_time} seconds."

File: C#/problem1.cs
using System;

class Problem1
{
  public static void Main()
  {
    int n = 1000;
    int m1 = 3;
    int m2 = 5;
    int sum = 0;
    for (int i = 1; i < n; i++)
    {
      if (i % m1 == 0 || i % m2 == 0)
      {
        sum += i;
      }
    }
    Console.WriteLine(sum);
  }
}
